### DISCUSSION TITLE
0042 XLS 42d: XRPL Plugins

### DISCUSSION THREAD

**Discussion by mvadari** [XRPLF/XRPL-Standards#116] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by dangell7** [comment]
I like this and think its great!

Obviously the security concern about MITM or injection of code but you made a point of that and labeled it as experimental so I'm excited to try it!

I feel like with these new "smart contract" features we should create a guide to help developers understand when a certain functionality might be right for their use case. "When to use... When not to use..." Nothing crazy

**Reply by mvadari** [reply]
MITM and code injection will be difficult since there's no network calls (outside of peering/consensus).

Re the guide: I have some notes on this topic, which will probably eventually be turned into a more formal blog post/guide.

**Reply by wojake** [reply]
Any updates on this @mvadari ?

**Reply by mvadari** [reply]
On what? @wojake

**Reply by wojake** [reply]
On this @mvadari

**Reply by mvadari** [reply]
Which part?

**Reply by wojake** [reply]
Jokes aside, I really think this amendment is important for future development efforts. I was, initially, wondering if you could write more on MITM attack(s)

**Reply by mvadari** [reply]
My comment (and Denis's question) was two parted (attacks and guides), I wasn't sure which you were referring to.

My statement still stands. Everything is running locally on the node/validator operator's server (plugins are connected at runtime via dynamic libraries with a link to the dylib in the `rippled.cfg` file), there is no additional network connection with plugins unless the plugin dev adds one. Certainly someone who has a better understanding of security than me should take a look at the architecture, but I don't think there's a possibility for MITM/injection attacks - at least, any more than the chances are right now.

**Comment by MarvinSunday** [comment]
Great work @mvadari 
I'm a big fan...
Love the work...

**Comment by mvadari** [comment]
Hello group: per the new [XLS Contributing process](https://github.com/XRPLF/XRPL-Standards/blob/master/CONTRIBUTING.md), it is my opinion that we have reach a "well-refined standard."

As such, I propose that we move this discussion to a file (via https://github.com/XRPLF/XRPL-Standards/pull/189) and work on final changes using additional PRs, for better change-tracking.

Please comment here if you would like to object to moving this spec/discussion forward in the process into a DRAFT spec.

(Note that per the Contributing guidelines, moving a spec into the [DRAFT](https://github.com/XRPLF/XRPL-Standards/blob/master/CONTRIBUTING.md#drafts) state does not mean any kind of endorsement, nor does it mean that this specification will become adopted. It is solely meant as a mechanism to enable better change tracking using PRs.)

**Comment by sappenin** [comment]
Closing this discussion since the spec has been initially merged via https://github.com/XRPLF/XRPL-Standards/pull/189.

For further discussion or changes, please open a PR on XLS-42d.
